S3 tests fail with node 16

The [tests which use S3](https://github.com/mapbox/node-pre-gyp/blob/master/test/build.test.js#L280) to publish and unpublish binaries weren't running on travis for the past few months. They can be activated currently when mock s3 flag is toggled.
"app can be installed via remote" test fails with node-gyp rebuild errors on node16 but passes with all older versions. The failure is not necessarily related to mock s3 library as they also fail with the same error when a dummy aws access key is provided in the variable.
This needs investigation and a fix.

Env: observed on travis jobs and local ubuntu bionic container
Node version : 16

### Error
1. The error from the tests is: 
`not ok 35 Error: Command failed 'V=1 npm install --clang=1 --fallback-to-build=false'` 

2. Stack shows `node-gyp` trying to rebuild something other than the test app. 
`npm ERR! c++ -o Release/obj.target/zmq/binding.o ../binding.cc '-DNODE_GYP_MODULE_NAME=zmq' '-DUSING_UV_SHARED=1' '-DUSING_V8_SHARED=1' '-DV8_DEPRECATION_WARNINGS=1' '-DV8_DEPRECATION_WARNINGS' '-DV8_IMMINENT_DEPRECATION_WARNINGS' '-D_GLIBCXX_USE_CXX11_ABI=1' '-D_DARWIN_USE_64_BIT_INODE=1' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DOPENSSL_NO_PINSHARED' '-DOPENSSL_THREADS' '-DBUILDING_NODE_EXTENSION' -I/Users/ron.ilan/Library/Caches/node-gyp/16.13.0/include/node -I/Users/ron.ilan/Library/Caches/node-gyp/16.13.0/src -I/Users/ron.ilan/Library/Caches/node-gyp/16.13.0/deps/openssl/config -I/Users/ron.ilan/Library/Caches/node-gyp/16.13.0/deps/openssl/openssl/include -I/Users/ron.ilan/Library/Caches/node-gyp/16.13.0/deps/uv/include -I/Users/ron.ilan/Library/Caches/node-gyp/16.13.0/deps/zlib -I/Users/ron.ilan/Library/Caches/node-gyp/16.13.0/deps/v8/include -I/opt/local/include  -O3 -gdwarf-2 -mmacosx-version-min=10.13 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++14 -stdlib=libc++ -fno-rtti -fno-strict-aliasing -MMD -MF ./Release/.deps/Release/obj.target/zmq/binding.o.d.raw   -c`

### Analysis 
1. The issue appears in `v15.0.0 (npm v7.0.2)` and above and does not appear in `v14.19.1 (npm v6.14.16)` and below. It is most likely `npm` version rather than node version related. 
2. Going one level deeper `npm v7.0.2` came with [`"node-gyp": "^7.1.1"`](https://github.com/npm/cli/blob/v7.0.2/package.json#L84) and
`npm v6.14.16` came with [`"node-gyp": "^5.1.0"`](https://github.com/npm/cli/blob/v6.14.16/package.json#L96) so it is most likely `node-gyp` related.

### Fix
Issue can be resolved by removing the always inserted `--clang=1` flag  from the [test command](https://github.com/mapbox/node-pre-gyp/blob/master/test/run.util.js#L62).

--- a/.travis.yml
+++ b/.travis.yml
@@ -42,7 +42,6 @@ matrix:
       node_js: 16
       env:
         - node_pre_gyp_mock_s3=true
-        - bypass_s3_tests=true
       addons:
         apt:
           sources: ['ubuntu-toolchain-r-test']

--- a/test/build.test.js
+++ b/test/build.test.js
@@ -277,9 +277,7 @@ apps.forEach((app) => {
   });
 
   const env = process.env;
-  // bypass_s3_tests can be used to avoid these tests for node 16 temporarily to unblock travis CI.
-  // See https://github.com/mapbox/node-pre-gyp/issues/638
-  if (!env.bypass_s3_tests && (env.AWS_ACCESS_KEY_ID || env.node_pre_gyp_accessKeyId || env.node_pre_gyp_mock_s3)) {
+  if (env.AWS_ACCESS_KEY_ID || env.node_pre_gyp_accessKeyId || env.node_pre_gyp_mock_s3) {
 
     test(app.name + ' publishes ' + app.args, (t) => {
       run('node-pre-gyp', 'unpublish publish', '', app, {}, (err, stdout) => {

--- a/test/run.util.js
+++ b/test/run.util.js
@@ -57,9 +57,6 @@ function run(prog, command, args, app, opts, cb) {
   if (!opts.cwd) {
     opts.cwd = path.join(__dirname, app.name);
   }
-  // avoid breakage when compiling with clang++ and node v0.10.x
-  // This is harmless to add for other versions and platforms
-  final_cmd += ' --clang=1';
 
   // Test building with msvs 2015 since that is more edge case than 2013
   if (process.platform === 'win32') {

